packages/charts/src/components/legend/legend_icon.tsx

```diff
@@ -9,36 +9,43 @@
 import React from 'react';
 
 import { ShapeRendererFn } from '../../chart_types/xy_chart/renderer/shapes_paths';
-import { Color } from '../../utils/common';
-import { PointStyle } from '../../utils/themes/theme';
-import { Icon } from '../icons/icon';
+import { Color, isNil } from '../../utils/common';
+import { PointShape, PointStyle } from '../../utils/themes/theme';
 
 interface LegendIconProps {
   pointStyle?: PointStyle;
   color: Color;
 }
 
-const MARKER_SIZE = 16;
+const MARKER_SIZE = 8;
+
+// to limit size, set a min and max
+const getRatio = (radius: number | undefined, strokeWidth: number) => {
+  const adjustedRadius = isNil(radius) ? strokeWidth : ((MARKER_SIZE / 2) * strokeWidth) / (radius * 2);
+  if (1.5 > adjustedRadius) return 1.5;
+  return Math.min(adjustedRadius, (radius ?? strokeWidth) * 2);
+};
+
 /** @internal */
 export const LegendIcon = ({ pointStyle, color }: LegendIconProps) => {
-  if (!pointStyle || !pointStyle.shape) {
-    return <Icon type="dot" color={color} aria-label={`series color: ${color}`} />;
-  }
-  const { shape, fill, stroke, strokeWidth, opacity } = pointStyle;
-  const [shapeFn, rotation] = ShapeRendererFn[shape];
-  const adjustedSize = MARKER_SIZE - strokeWidth;
+  const { shape, fill, stroke, strokeWidth = 1, opacity, radius } = pointStyle ?? { radius: 4 };
+
+  const adjustedFill = shape ? fill ?? color : color;
+  const adjustedStrokeWidth = getRatio(radius, strokeWidth);
+  const [shapeFn, rotation] = ShapeRendererFn[shape ?? PointShape.Circle];
+  const adjustedSize = MARKER_SIZE - adjustedStrokeWidth;
   return (
-    <svg height={MARKER_SIZE} width={MARKER_SIZE} aria-label={`series color: ${color}`}>
+    <svg width={MARKER_SIZE * 2} height={MARKER_SIZE * 2} viewBox="0 0 16 16" aria-label={`series color: ${color}`}>
       <g
         transform={`
-            translate(${MARKER_SIZE / 2}, ${MARKER_SIZE / 2})
+            translate(${MARKER_SIZE}, ${MARKER_SIZE})
             rotate(${rotation})`}
       >
         <path
-          d={shapeFn(shape === 'triangle' || shape === 'plus' || shape === 'x' ? adjustedSize / 2 : adjustedSize / 3)}
+          d={pointStyle?.shape === PointShape.Diamond ? shapeFn(adjustedSize / 3) : shapeFn(adjustedSize / 2)}
```

Comment:
What is this for? We shouldn't divide by `3` cuz that's just arbitrary. Some shapes may appear bigger than others but they are all still the same size.

```suggestion
          d={shapeFn(adjustedSize / 2)}
```

Reply:
The diamonds only have cutoff but the other shapes seem fine with `/2`

Reply:
can you show me a pic of what you mean?

Reply:
Ok I see what you mean here. I think a better solution would be to scale down the diamond to fit inside of the "viewport". Like described [here](https://math.stackexchange.com/a/60722/952008). 

![image](https://user-images.githubusercontent.com/19007109/126541924-1e5a67d0-269d-4e10-b959-c3041851ca56.png)

### Before
![Image 2021-07-21 at 12 57 13 PM](https://user-images.githubusercontent.com/19007109/126542015-3f44975c-7394-4619-bed7-7798d2803c12.jpg)

### After
![Image 2021-07-21 at 1 16 54 PM](https://user-images.githubusercontent.com/19007109/126542031-b62743dc-66d8-4fdb-8c17-aec3f98d6a59.jpg)

> Notice the diamond is now contained inside the square. See code changes in 8637423b929d3ee55f2e7d3ba8b556f4a605d45a.